### PR TITLE
Todo 7

### PR DESCRIPTION
**Pull request by Oleg116**
Here is filter added, please check it and let me know what should be better

### CODE REVIEW

File: src/ToDo.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import style from './Todo.module.scss';
+
+const ToDo = ({
+  toDoList, addToDo, removeToDo, clearCompleted, makeDone,
+  setAllCompleted, setAllUncompleted, showCompleted, showActive, showAll,
+}) => {
+  const left = toDoList.filter(item => !item.isDone).length;
```
Comment (jestkijowngnax) [inline src/ToDo.js:8]: bad name for a function
what is`left`? You should named function depends which this function do.

File: src/ToDo.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  const addInputValue = e => {
```
Comment (jestkijowngnax) [inline src/ToDo.js:9]: please use the same names of arguments, if you write **event** lets do this for all func which you use events.

File: src/ToDo.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    if (e.key === 'Enter' && e.target.value !== '') {
+      addToDo(e.target.value);
+      e.target.value = '';
+    }
+  };
+  const removeToDoItem = (event) => {
+    const shouldRemove = Number(event.target.parentNode.children[1].classList.value);
```
Comment (jestkijowngnax) [inline src/ToDo.js:16]: why you use parentNode? 
I told you to remove item by id, please do this.

File: src/ToDo.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    removeToDo(shouldRemove);
+  };
+
+  const filter = (event, reduxFunc) => {
+    for (let i = 1; i <= 3; i++) {
+      event.target.parentNode.children[i].disabled = false;
```
Comment (jestkijowngnax) [inline src/ToDo.js:22]: don`t use it!!!

File: src/ToDo.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      e.target.value = '';
+    }
+  };
+  const removeToDoItem = (event) => {
+    const shouldRemove = Number(event.target.parentNode.children[1].classList.value);
+    removeToDo(shouldRemove);
+  };
+
+  const filter = (event, reduxFunc) => {
+    for (let i = 1; i <= 3; i++) {
```
Comment (jestkijowngnax) [inline src/ToDo.js:21]: why you use for loop? you have map, or forEach

File: src/ToDo.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      addToDo(e.target.value);
+      e.target.value = '';
+    }
+  };
+  const removeToDoItem = (event) => {
+    const shouldRemove = Number(event.target.parentNode.children[1].classList.value);
+    removeToDo(shouldRemove);
+  };
+
+  const filter = (event, reduxFunc) => {
```
Comment (jestkijowngnax) [inline src/ToDo.js:20]: reduxFunc? bad naming
this is not clear name.
change it please.

File: src/ToDo.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    for (let i = 1; i <= 3; i++) {
+      event.target.parentNode.children[i].disabled = false;
+    }
+    event.target.disabled = true;
+    reduxFunc();
+  };
+
+  const completeToggle = () => {
+    for (let i = 0; i < toDoList.length; i++) {
```
Comment (jestkijowngnax) [inline src/ToDo.js:29]: don`t use for loop!!!

File: src/ToDo.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+      if (!toDoList[i].isDone) {
+        setAllCompleted();
+        break;
```
Comment (jestkijowngnax) [inline src/ToDo.js:32]: do you need break??

File: src/ToDo.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        setAllUncompleted();
+      }
+    }
+  };
+
+  return (
+    <div className={style.todoApp}>
+      <p className={style.jumbotron}>todos</p>
+      <div className={style.inputField}>
+        <button className={style.setAll} type="button" onClick={() => completeToggle()}>{'>'}</button>
```
Comment (jestkijowngnax) [inline src/ToDo.js:43]: you do not need arrow func and call it. just write onClick={completeToggle}

File: src/ToDo.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+        <input className={style.input} type="text" onKeyPress={addInputValue} placeholder="What needs to be done?" />
+      </div>
+      <div className={style.toDoList}>
+        <ul>
+          {toDoList.map((element, index) => (
+            <li className={style.todoItem} key={index.toString()}>
```
Comment (jestkijowngnax) [inline src/ToDo.js:49]: why you convert it to string?

File: src/ToDo.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+              <input className={style.resultBox} checked={toDoList[index].isDone} type="checkbox" onChange={(e) => makeDone(Number(e.target.value))} value={index} />
```
Comment (jestkijowngnax) [inline src/ToDo.js:50]: why Number?

File: src/ToDo.js [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+              <span
+                className={index}
+              >
+                {element.name}
+              </span>
+              <button className={style.deleteSpec} type="button" onClick={removeToDoItem}>X</button>
```
Comment (jestkijowngnax) [inline src/ToDo.js:56]: please use svg icon to indicate closing button.

File: src/ToDo.js [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+            </li>
+          ))}
+        </ul>
+        <div className={style.buttonsContainer}>
+          <p className={style.left}>
+            {`${left} `}
```
Comment (jestkijowngnax) [inline src/ToDo.js:62]: you don`t need templste string, just write {left}

File: src/ToDo.js [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+            Items left
+          </p>
+          <button id="all" type="button" onClick={() => filter(event, showAll)}>All</button>
```
Comment (jestkijowngnax) [inline src/ToDo.js:65]: where is your event?

File: src/Todo.module.scss [hunk 15]
```diff
@@ -0,0 +1,147 @@
+$base-color: #fcfcfc;
```
Comment (jestkijowngnax) [inline src/Todo.module.scss:1]: move all variables to additional folder with all variables.

File: src/store/counterData/reducer.js [hunk 16]
```diff
@@ -0,0 +1,99 @@
+const initialState = {
+  toDo: [],
+  toDoBoofer: [],
```
Comment (jestkijowngnax) [inline src/store/counterData/reducer.js:3]: you don`t need the same array, work with first one.

File: src/store/counterData/reducer.js [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+              name: element.name,
+              isDone: !element.isDone,
+            };
+          }
+          return {
+            name: element.name,
+            isDone: element.isDone,
+          };
+        }),
+        toDoBoofer: state.toDo.map((element, index) => {
```
Comment (jestkijowngnax) [inline src/store/counterData/reducer.js:41]: I told you to create each item with **value, id, status**, and remove it only by current id.
Status should be changed only by actions, and will be filtered only one array of items.
Why you ignore that?

remove all cases, and write it correctly.
addToDo, removeToDo, etc.

**Review by jestkijowngnax — CHANGES_REQUESTED** [review]

File: src/ToDo.js (hunk 8 above)
Comment (Oleg116) [reply src/ToDo.js:32]: I do

File: src/ToDo.js (hunk 11 above)
Comment (Oleg116) [reply src/ToDo.js:50]: Unfortunately, couse this action can work only with numbers

**Pull request merged**